### ISSUE TITLE
implement resumable indexing

### ISSUE THREAD

**Issue by boudra** [gitcoinco/grants-stack-indexer#316]
We might not need to do this if the depoyments become reliable and fast enough.

from @bard:

>  Separate instances make it tricky to do anything like setting a flag in a db or on the filesystem. Maybe the build tag could be used in conjunction with Fly secrets, i.e. fly secret set REINDEX_AT_REV=0a586b --staged and then in the code if (config.buildTag === config.reindexAtStage) { reindex() }

**Comment by boudra** [comment]
@bard what do you think about this approach:

1. We introduce the concept of a "data version number", we could put it in the codebase as a hard coded value or as an environment variable that we modify directly in the fly.toml, so this version number is coupled to the codebase, when we make a schema change we increment this number
2. Each instance tracks the currently indexed version in a file, if the version indexed in disk is different from the one that we're currently running, we delete the chain directory and reindex

Basically the same approach you proposed but we track the version with the code instead of in a secret, this way we can review it in PRs and there's no "manual step", there's still the chance that we forget to change this number but I think it will be hard to come up with a solution where that is not an issue.

**Comment by boudra** [comment]
Also from our other conversation in Notion, I agree that we can skip doing SQL migrations and just use whatever system we come up here to reindex instead of migrating since all data we have is always sourced from elsewhere.

**Comment by bard** [comment]
> @bard what do you think about this approach:

Yeah, this sounds much better, is a battle-tested approach and keeps a paper trail in git.
